test/test_main.py

```diff
@@ -73,7 +74,7 @@ def __init__(self, event_type, event_name, env_var=None,
         self.charm_config = charm_config
 
 
-class TestMain(unittest.TestCase):
+class TestMain(abc.ABC):
```

Comment:
why ABC?

Reply:
my intent was to signal that this class was not meant to be instantiated, as some methods need to be implemented still. I should probably use the `@abstractmethod` thing to highlight which though (or drop the ABC entirely)

Reply:
Would it be better to change the name of it? TestMainFixture? MainTestMixin? I don't have great names for it, but something like Fixture or Mixin signal to me that it isn't actually going to be tested by itself.